### PR TITLE
fix bootstrapping process in the session mode

### LINKED ISSUE #2863: Fix Session logic for the bootstrapping process

### PR DESCRIPTION
**Pull request by chanwit**
This PR fixes the bootstrap process after closing a session by:
  - Making GitOps Run aware of bootstrapping after closing a session
  - Adding `--dashboard-hashed-password`. The outer process is now responsible for asking the dashboard question and it passes the hashed password into the inner process.

Fixes #2863

### CODE REVIEW

**Comment by opudrovs** [comment]
I'm getting a weird empty error message on Ctrl+C if refusing bootstrap (not sure if it's related to this PR or bootstrap code in general):

```
Handling connection for 9898
^C

► Deleting GitOps Run session run-master-e40d32ba-dirty ...
^C✔ Session run-master-e40d32ba-dirty is deleted successfully
✗ Would you like to bootstrap your cluster into GitOps mode?: 
Error: 
olga@macbook-pro podinfo % 
```

File: cmd/gitops/beta/run/cmd.go [hunk 1]
```diff
@@ -283,18 +287,108 @@ func installFlux(log logger.Logger, kubeClient *kube.KubeHTTP) (fluxVersion stri
...
+			return "", false, err
 		}
 
 		fluxVersion = flags.FluxVersion
 	} else {
 		log.Successf("Flux version %s is found", fluxVersion)
 	}
 
-	return fluxVersion, nil
+	return fluxVersion, true, nil
```
Comment (ozamosi) [inline cmd/gitops/beta/run/cmd.go:298]: This doesn't look right - the else branch needs to return justInstalled = false, doesn't it?

**Comment by opudrovs** [comment]
- I am not able to log into the dashboard when using the following command:

```
/Volumes/Macintosh_HD_Data/_job_ww/_src/weave-gitops-3/bin/gitops beta run ./deploy/overlays/dev --timeout=3m --port-forward namespace=dev,resource=svc/backend,port=9898:9898 --dashboard-hashed-password=$2a$10$7DoHm1w0R2HE4BsWqbs4cepIdjLC3ZdlIoh9djtbBRHRX0omFta7y
```

The password hash was generated with:

```
olga@macbook-pro ~ % PASSWORD="123"                     
echo -n $PASSWORD | gitops get bcrypt-hash
$2a$10$7DoHm1w0R2HE4BsWqbs4cepIdjLC3ZdlIoh9djtbBRHRX0omFta7y
```

I'm getting the following error when attempting to log in with `admin` and `123`:

Screenshot 2022-10-13 at 11 36 34


- There are two very similar pieces of code after the following comments:

```
// run bootstrap wizard only if Flux was not installed
```
Can they be combined into a function or smth. like that?

- Bootstrap prompts have two question marks. Is one question mark added automatically by promptui? Question marks should be removed from prompt labels (I've counted four prompts with question marks in the code but have not tested them all yet).

Screenshot 2022-10-13 at 11 38 04

**Comment by ozamosi** [comment]
Both promptui comments (returning the Aborted error, and the extra question mark) are from #2853, not this PR.

**Comment by opudrovs** [comment]
I've just tested that I can log into the dashboard successfully with `admin` and `123` if entering the password manually and not using the `--dashboard-hashed-password` flag.

**Comment by opudrovs** [comment]
Tested bootstrapping, it looks nice and the process is very smooth! 🎉 

The only minor issue is that I am getting emails about this in my repo. As far as I understand, the workflow was copied from podinfo. Probably expected.

Screenshot 2022-10-13 at 12 18 22

Screenshot 2022-10-13 at 12 18 44

**Comment by chanwit** [comment]
@opudrovs it's the shell problem, please try --dashboard-hashed-password='$2a$10$7DoHm1w0R2HE4BsWqbs4cepIdjLC3ZdlIoh9djtbBRHRX0omFta7y'

```
❯ echo $2a$10$7DoHm1w0R2HE4BsWqbs4cepIdjLC3ZdlIoh9djtbBRHRX0omFta7y
a0DoHm1w0R2HE4BsWqbs4cepIdjLC3ZdlIoh9djtbBRHRX0omFta7y

❯ echo "$2a$10$7DoHm1w0R2HE4BsWqbs4cepIdjLC3ZdlIoh9djtbBRHRX0omFta7y"
a0DoHm1w0R2HE4BsWqbs4cepIdjLC3ZdlIoh9djtbBRHRX0omFta7y

❯ echo '$2a$10$7DoHm1w0R2HE4BsWqbs4cepIdjLC3ZdlIoh9djtbBRHRX0omFta7y'
$2a$10$7DoHm1w0R2HE4BsWqbs4cepIdjLC3ZdlIoh9djtbBRHRX0omFta7y
```

**Review by ozamosi — APPROVED** [review]

**Comment by opudrovs** [comment]
@chanwit thanks, it worked! I We need to let the user know about single quotes then. tried double quotes before and it didn't help, didn't know they were supposed to be single quotes.

**Review by opudrovs — APPROVED** [review]

**Comment by opudrovs** [comment]
@chanwit If this flag is used exclusively for communication between processes it would be nice to mark it as no-op in the PR description and/or flag description. Because the description says it is used for passing hashed password between processes but does not say the user should not use it directly. It is misleading. Because if it is a public flag it can be used directly by definition. That's why I used it for testing.

**Pull request merged**